src/analyze.rs

```diff
@@ -167,13 +167,13 @@ impl<'a> Analyzer<'a> {
             }
             Expr::Backref(group) => {
                 if group >= self.group_ix {
-                    return Err(Error::InvalidBackref);
+                    return Err(Error::InvalidBackref(0));
```

Comment:
I wasn't sure what position to return when we return an error during analyze or compile time which can also occur at parse time.

Reply:
fixed by adding separate error types for parse time errors, analyze/compile time errors and runtime errors